homeassistant/components/history.py

```diff
@@ -7,16 +7,37 @@
 from collections import defaultdict
 from datetime import timedelta
 from itertools import groupby
+import voluptuous as vol
 
+import homeassistant.helpers.config_validation as cv
 import homeassistant.util.dt as dt_util
 from homeassistant.components import recorder, script
 from homeassistant.components.frontend import register_built_in_panel
 from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import ATTR_HIDDEN
 
 DOMAIN = 'history'
 DEPENDENCIES = ['recorder', 'http']
 
-SIGNIFICANT_DOMAINS = ('thermostat',)
+CONF_EXCLUDE = 'exclude'
+CONF_INCLUDE = 'include'
+CONF_ENTITIES = 'entities'
+CONF_DOMAINS = 'domains'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        CONF_EXCLUDE: vol.Schema({
+            vol.Optional(CONF_ENTITIES, default=[]): cv.ensure_list,
+            vol.Optional(CONF_DOMAINS, default=[]): cv.ensure_list
+        }),
+        CONF_INCLUDE: vol.Schema({
+            vol.Optional(CONF_ENTITIES, default=[]): cv.ensure_list,
```

Comment:
If you default it to an empty list, you are unable to differentiate between not getting a value (`None`) or a user wanting to filter out everything (`[]`).